manifests/value.pp

```diff
@@ -69,13 +70,13 @@
   Registry_key { ensure => present }
   Registry_value { ensure => present }
 
-  if !defined(Registry_key[$key]) {
-    registry_key { $key: }
+  if !defined(Registry_key[$key_normalized]) {
+    registry_key { $key_normalized: }
```

Comment:
It it defines a `registry_value` with the original case, there's no way to reliably check to see if it's already defined, unless we write a new `case_insensitive_defined()` function.